### DISCUSSION TITLE
Method not supported error when accessing '/' with routes cached

### DISCUSSION THREAD

**Discussion by susseau** [laravel/framework#34802] [Q&A]
- Laravel Version: 7.28.4
- PHP Version: 7.4.3
- Database Driver & Version:

### Description:
Browsing to '/' produces error:

> Symfony\Component\HttpKernel\Exception\MethodNotAllowedHttpException
> The GET method is not supported for this route. Supported methods: HEAD.

### Steps To Reproduce:
Put below code into routes/web.php. May be any other controller, etc. Execute php artisan route:cache. Without caching it works fine.

```
<?php

Route::middleware(['auth'])->group(function () {
    Route::get('/', 'RedirectController');
});
```

There seems to be a problem with requestWithoutTrailingSlash in Illuminate/Routing/CompiledRouteCollection.
It's described here https://laracasts.com/discuss/channels/code-review/laravel-7-breaks-my-index-route-whenever-the-routes-are-cached
Removing rtrim function helps as described in link above.

**Comment by rodrigopedra** [comment]
Is your Laravel root path within a subdirectory in your domain?

Something like this: https://example.com/my-app/

If so, subdirectory installs are not supported:

https://laravel.com/docs/8.x/installation#directory-configuration

There are several closed issues regarding this, I remember on reading people complaining route caching didn't work with that setup and core team members noting installing a Laravel app to be served from a subdirectory not being supported.
